src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -87,7 +87,7 @@ def _list(ioctx):
                     result = []
             else:  # pool config
                 pg_status = list(CephService.get_pool_pg_status(self._pool_name).keys())
-                if len(pg_status) == 1 and 'unknown' in pg_status[0]:
+                if len(pg_status) == 1 and 'incomplete' in pg_status[0]:
```

Comment:
This change will unfix the backported fix. If this get's merged the dashboard will not respond if the profile of a pool doesn't meet the spec of your cluster, as the `incomplete` flag is used in newer versions only, at least I'm not aware of any change since my backport.

Reply:
@Devp00l Could you please elaborate what you mean that the `incomplete` flag is used in newer versions only?

'incomplete' is a pg state and from the code I don't see why it may not be present in `pg_status` returned by `CephService.get_pool_pg_status` in nautilus (if the pool really has some pgs in this state).

Reply:
The flag is not set for the case that an ec profile is in use that is configured to use more failure domains than available or failure domains that do not exist. For these pools octopus and pacific show the pg_status `incomplete` where in nautilus the pg_status `unknown` is set. Anyway which state is shown here the problem is that `_rbd.config_list` won't return any result in that case and the dashboard got stuck. For details take a look at the introducing PR #32829 and it's issue. Also take a look at the comment down below in the file.

This line will not throw any error in the case of an unknown state it will just not call the config_list method and return an empty array. If no test failure is saying expected something else than an empty array, this is not the right spot to fix the jenkins failure.

I also want to remind everyone that this error could be also have bubbled up from a totally different area in ceph, but I still think it has something to do with our upstream test environment, as I couldn't reproduce any pool creation failure locally.

Reply:
Yes, and as I wrote in the tracker ticket [1] the initial approach to fix this issue with `rbd.config_list` potentially getting stuck for a non-healthy pool is incorrect.

You may have a million reasons why the pool is a non healthy state, which may case rados operations to this pool to get stuck. You observed one case in your environment and "fixed" for this case  making the thing more confusing for other cases. Potentially, just one pg in the pool in non-active state may cause `rbd.config_list` to get stuck. So if you really want to make sure it will not get stuck, you have to check that all pgs are in active state. And return the error so the caller is not confused.

Usually the 'unknown' state is a transient state. When you create a new pool, all pgs are in 'unknown' state until they are created and activated by osds.

What is going here I suppose is that the test creates a new pool (pgs are in unknown state initially), starts copying configuration at the new pool and at that moment gets stuck (because the pool is not activated yet), then when pgs are active the test proceeds and when tries to read the configuration back it hits your condition. And here we have a race: the pg stats may already be updated in mgr or not, and if it is not (stats still reports pgs in 'unknown' state) the empty list is returned.

[1] https://tracker.ceph.com/issues/46815#note-29